common/src/main/java/com/wynntils/features/DiscordRichPresenceFeature.java

```diff
@@ -126,42 +126,43 @@ private void checkTerritory() {
     }
 
     @Override
-    protected void onConfigUpdate(Config<?> config) {
-        if (this.isEnabled()) {
-            // This isReady() check is required for Linux to not crash on config change.
-            if (!Services.Discord.isReady()) {
-                // Even though this is in the onConfigUpdate method, it is how the library is first loaded on launch
-                if (!Services.Discord.load()) {
-                    // happens when wrong version of GLIBC is installed and Discord SDK fails to load
-                    Managers.Feature.crashFeature(this);
-                }
+    public void onEnable() {
+        // This isReady() check is required for Linux to not crash on config change.
+        if (!Services.Discord.isReady()) {
+            // Even though this is in the onConfigUpdate method, it is how the library is first loaded on launch
+            if (!Services.Discord.load()) {
+                // happens when wrong version of GLIBC is installed and Discord SDK fails to load
+                Managers.Feature.crashFeature(this);
             }
+        }
 
-            if (!Models.WorldState.onWorld() && Services.Discord.isReady()) return;
+        if (!Models.WorldState.onWorld() && Services.Discord.isReady()) return;
```

Comment:
I'm trying to make sense of this.

1) If enabled at startup, onWorld is always false, so this returns if discord is ready, otherwise it continues below.

2) If enabled later, while in the hub, and discord is not ready, it continues below.

Both of these behaviors seems sus.

Reply:
Most likely I have to separate the enabling/disabling logic from the config update method, and add it back (and also keep enable/disable separate). Overall, the aim is to reduce the native library calls, as they are slow.

Reply:
Should be fixed now.

Reply:
I still think it looks weird. Are you sure you don't mean:
```
 if (!(Models.WorldState.onWorld() && Services.Discord.isReady())) return;
 ```
?

Reply:
Probably should be an or. Not on world, or discord is not ready.

Reply:
If so, that is prhaps better expressed as:
```
 if (!Models.WorldState.onWorld()) return;
 if (!Services.Discord.isReady()) return;
```